debug/connection-channel.js

```diff
@@ -0,0 +1,80 @@
+/**
```

Comment:
what is the debug/ directory? could this be runtime/debug?

Reply:
Yep, I was considering this. Moved.